frontend/src/components/scan/utils/RuntimeConfigurationModal.jsx

```diff
@@ -7,123 +7,137 @@ import { ContentSection, CustomJsonInput } from "@certego/certego-ui";
-          // For each param (dict) extract the value of the "value" key
-          [pluginName]: Object.entries(pluginParams).reduce(
-            (singlePluginConfig, [paramName, { value: paramValue }]) => ({
-              ...singlePluginConfig,
-              [paramName]: paramValue,
-            }),
-            {},
-          ),
-        }),
-        {},
-      ),
-    [combinedParamsMap],
-  );
+    };
+  });
 
-  console.debug("RuntimeConfigurationModal - defaultNameParamsMap:");
-  console.debug(defaultNameParamsMap);
+  console.debug("RuntimeConfigurationModal - selectedPluginsParams:");
+  console.debug(selectedPluginsParams);
 
   /* this is the dict shown when the modal is open: load the default params and the previous saved config
     (in case the user update the config, save and close and reopen the modal)
-
-    IMPORTANT: We want to group the plugins in the categories (analyzers, etc...), it is more difficult to handle it in every variable
-    so we use it only when shown to the user (UI), we need to rembeber in case we edite the params more than once we need to 
-    load the data from the structure with the categories.
   */
-  const editableConfig = React.useMemo(() => {
-    const config = {
-      ...defaultNameParamsMap,
-      ...(formik.values.runtime_configuration.analyzers || {}), // previous values of analyzers (groupped by the previous editing) if present
-      ...(formik.values.runtime_configuration.connectors || {}), // previous values of connectors (groupped by the previous editing) if present
-    };
-    const analyzerNames = analyzers.map((analyzer) => analyzer.name);
-    const connectorNames = connectors.map((connector) => connector.name);
-    const result = { analyzers: {}, connectors: {} };
-    Object.entries(config).forEach(([configPluginName, configPluginParams]) => {
-      if (analyzerNames.includes(configPluginName)) {
-        result.analyzers[configPluginName] = configPluginParams;
-      } else if (connectorNames.includes(configPluginName)) {
-        result.connectors[configPluginName] = configPluginParams;
-      }
-    });
-    return result;
-  }, [
-    analyzers,
-    connectors,
-    defaultNameParamsMap,
-    formik.values.runtime_configuration.analyzers,
-    formik.values.runtime_configuration.connectors,
-  ]);
+  Object.keys(selectedPluginsInFormik).forEach((pluginType) => {
+    editableConfig[pluginType] = Object.entries(
+      selectedPluginsParams[pluginType],
+    ).reduce(
+      (generalConfig, [pluginName, pluginParams]) => ({
+        ...generalConfig,
+        // For each param (dict) extract the value of the "value" key
+        [pluginName]: Object.entries(pluginParams).reduce(
+          (singlePluginConfig, [paramName, { value: paramValue }]) => ({
+            ...singlePluginConfig,
+            [paramName]: paramValue,
+          }),
+          {},
+        ),
+        ...(formik.values.runtime_configuration[pluginType] || {}),
+      }),
+      {},
+    );
+  });
```

Comment:
it's correct but I think a reduce inside another reduce is quite difficult to read, I suggest to use forEach instead, I hope it will be more readable.